### ISSUE TITLE
PyTorch model is not converging 

### ISSUE THREAD

**Issue by Aexolowski** [DeepLabCut/DeepLabCut#2894]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Bug description

Pytorch_config.yaml file: dataloader_pin_memory is default set to "false" which results in pose prediction errors when trained on a NVIDIA GPU

### Operating System

Windows 11

### DeepLabCut version

3.0.0rc6

### DeepLabCut mode

single animal

### Device type

NVIDIA RTX3080

### Steps To Reproduce

Going throught he standard steps of pose prediction using the GUI

### Relevant log output

```shell

```

### Anything else?

The dataloader_pin_memory is default set to "false" and, assuming that most people use a NVIDIA GPU, this causes the learning_stats to have only "nan" values for test.rsme and test.rsme_pcutoff, and the evaluation results to have huge train and test rmse and no values for train and test rmse pcutoff, and when analysing a new video with this model, the tracked keypoints have the same x and y data and likelihoods for each frame.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by n-poulsen** [comment]
Hi @Aexolowski! The `dataloader_pin_memory` parameter simply links to the [`pin_memory`](https://pytorch.org/docs/stable/data.html#memory-pinning) parameter for PyTorch DataLoaders. We set the default to `False` as that's also the default value for the PyTorch DataLoaders.

It should have no impact at all on training performance - so if you get `NaN` values when evaluating, I'd check for errors in the ground truth data (with the [`check_labels`](https://deeplabcut.github.io/DeepLabCut/docs/maDLC_UserGuide.html#check-annotated-frames) function), or at editing other [training hyperparameters (such as the learning rate)](https://deeplabcut.github.io/DeepLabCut/docs/pytorch/pytorch_config.html#runner).

Is your loss going down "normally"? Would you perhaps have some metrics from the `learning_stats.csv` file or `train.txt` to share so I could take a look?

**Comment by Aexolowski** [comment]
Hi @n-poulsen !

Thanks for your reply! 
I checked mhy ground truth data and they seem to be fine. I attached the learning_stats.csv and train.txt from a project where I ran into that issue. I did not try to change training hyperparameters so far - could that be causing the issue in my case? Because when I tried to change the dataloader_pin_memory to TRUE I didn't see the nan values anymore.

[learning_stats.csv](https://github.com/user-attachments/files/18913913/learning_stats.csv)

[train.txt](https://github.com/user-attachments/files/18913915/train.txt)

**Comment by n-poulsen** [comment]
@Aexolowski thanks for the files! The loss doesn't seem to go down at all (it's really flat), suggesting that the model isn't learning. This may be an issue with the data, or hyperparameters. Could you try to do the following changing the batch size to 8 in your `pytorch_config.yaml` file?

And if you set dataloader_pin_memory to TRUE could you share what the learning curves look like?

**Comment by Aexolowski** [comment]
Hi again @n-poulsen!

Thanks for looking into this! I stopped working on this model and created a new one that seems to be better. I attached the learning_stat.csv and train.txt files from the third iteration of this model, where I set the dataloader_pin_memory to TRUE. I didn't change the batchsize as my GPU is already pretty maxed out with batchsize 1.

[train.txt](https://github.com/user-attachments/files/18998941/train.txt)
[learning_stats.csv](https://github.com/user-attachments/files/18998940/learning_stats.csv)

**Comment by n-poulsen** [comment]
Thanks @Aexolowski - here the loss looks much closer to what I'd expect it to be. Can you confirm that training with exactly the same `pytorch_config.yaml` file but simply switching `dataloader_pin_memory: false` again leads to the model not converging (i.e. the loss stays flat)?

**Comment by Aexolowski** [comment]
Hi again!
I did not yet try to run another training with this model while setting the dataloader_pin_memory: false.
However, I tried to set up a new experiment and trained the network with dataloader_pin_memory: true, but ran into the same issue again: in the learning_stats there are only "nan" values for test.rsme and test.rsme_pcutoff, and the loss doesn't converge. I attached the evaluation_results, learning_stats, config.yml, pytorch_config.yml, and train.txt files. 
I am very confused now about what the issue here really is...

[DLC_Resnet50_DLC_EZMFeb19shuffle1_snapshot_200-results.csv](https://github.com/user-attachments/files/19047994/DLC_Resnet50_DLC_EZMFeb19shuffle1_snapshot_200-results.csv)
[learning_stats.csv](https://github.com/user-attachments/files/19047995/learning_stats.csv)
[train.txt](https://github.com/user-attachments/files/19047996/train.txt)

**Comment by n-poulsen** [comment]
Hi @Aexolowski - thanks for sharing these. I think most likely the issue comes from the data. Are all bodyparts visible and labeled in every image that you have? And how big are your images?

**Comment by Aexolowski** [comment]
Hi @n-poulsen ! 

Actually no, in this video set, not all bodyparts are visible in each image - that might be the issue then. How do I get around this?

**Comment by n-poulsen** [comment]
The model will learn when bodyparts are visible or not - so there should be no issue in training there. What is needed for the model to able to learn is consistency in labelling between frames (so if two frames look a lot like one another, they should be labeled consistently).

If you were having one shuffle (with `pin_memory: True`) where the model was converging and others weren't, it may be that some mislabeled images were in the test set in one case and the training set in the other. What I would do is:

1. Use `check labels` to try and find any mislabeled images (swaps between left/right bodyparts are some of the most common)
2. If no incorrectly labeled images are found, create a few different shuffles (only training on part of the data) and see if some of them train correctly and others don't.

**Comment by Aexolowski** [comment]
Hi again! 
Okay, that makes sense, that could have been the issue! I will try out your suggestions and will get back in case I still run into the same issues! Thanks so much for your help, really appreciated!

**Comment by n-poulsen** [comment]
Thanks @Aexolowski - I'll close this for now, but re-open it if you run into the same issue
